the-collab-lab/tcl-78-smart-shopping-list
# Dc rt mark items

|     | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: New feature     |
|  ✓    | checked Item gets renewed after 24 hours |

## Description

firebase.js

We filled out the updateItem function asynchronously with parameters listPath, itemId and the dateLstPurchased and totalPurchases props. For this we referenced the item (i.e. document) in firestore to be updated.

We added a try-catch argument to update the document using the updateDoc function to update the dateLastPurchased and the totalPurchases made by the user. While the former takes in a date stamp, the latter adds the amount of times an item is bought.

ListItem.js

We included a useEffect hook to link the effects of ticking off a checkbox. Therefore, the change in State will trigger a change in the checkStatus, so that after 24 hours, the check reappears.

List.js

We passed the props to be updated in the ListItem component

## Related Issue

Closes #9 

## Acceptance Criteria

[x] The ListItem component renders a checkbox with a semantic <label>.
[x] Checking off the item in the UI also updates the dateLastPurchased and totalPurchases properties on the corresponding Firestore document
[x] The item is shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day). After 24 hours, the item unchecks itself so the user can buy it again.
[x] The updateItem function in firebase.js has been filled out, and sends updates to the firestore database when an item is checked

## Type of Changes

Added to and enhanced the check feature

## Updates

### Before

![Captura de pantalla 2024-09-07 a la(s) 15 02 52](https://github.com/user-attachments/assets/ed3c5e18-01bf-477c-ad1e-b7c0593e425b)


### After

![Captura de pantalla 2024-09-07 a la(s) 15 03 12](https://github.com/user-attachments/assets/3ae25128-d27e-4b7c-80c2-c221fc0d6a3f)


## Testing Steps / QA Criteria

1. npm start
2. Select a list and go to the List view
3. check the box off an item
4. item should be unchecked after 24 hours
